unit_test/docker_container_test.py

```diff
@@ -21,23 +25,178 @@
+    HAVE_PODMAN = bool(shutil.which("podman"))
+
+    if HAVE_DOCKER:
+        # Basic podman configuration
+        yield {"container_engine": "docker", "docker_image": DEFAULT_IMAGE}
+
+    if HAVE_PODMAN:
+        # Basic podman usage
+        yield {"container_engine": "podman", "docker_image": DEFAULT_IMAGE}
+
+        # VFS Podman usage (for the podman in docker use-case)
+        dpath = Path(temp_test_dir.name)
+
+        # This requires that we write configuration files and point to them
+        # with environment variables before we run podman
+        # https://github.com/containers/common/blob/main/docs/containers.conf.5.md
+        vfs_containers_conf_data = {
+            "containers": {
+                "default_capabilities": [
+                    "CHOWN",
+                    "DAC_OVERRIDE",
+                    "FOWNER",
+                    "FSETID",
+                    "KILL",
+                    "NET_BIND_SERVICE",
+                    "SETFCAP",
+                    "SETGID",
+                    "SETPCAP",
+                    "SETUID",
+                    "SYS_CHROOT",
+                ]
+            },
+            "engine": {"cgroup_manager": "cgroupfs", "events_logger": "file"},
+        }
+        # https://github.com/containers/storage/blob/main/docs/containers-storage.conf.5.md
+        storage_root = dpath / ".local/share/containers/vfs-storage"
+        run_root = dpath / ".local/share/containers/vfs-runroot"
+        storage_root.mkdir(parents=True, exist_ok=True)
+        run_root.mkdir(parents=True, exist_ok=True)
+        vfs_containers_storage_conf_data = {
+            "storage": {
+                "driver": "vfs",
+                "graphroot": str(storage_root),
+                "runroot": str(run_root),
+                "rootless_storage_path": str(storage_root),
+                "options": {
+                    # "remap-user": "containers",
+                    "aufs": {"mountopt": "rw"},
+                    "overlay": {"mountopt": "rw", "force_mask": "shared"},
+                    # "vfs": {"ignore_chown_errors": "true"},
+                },
+            }
+        }
+        vfs_containers_conf_fpath = dpath / "temp_vfs_containers.conf"
+        vfs_containers_storage_conf_fpath = dpath / "temp_vfs_containers_storage.conf"
+        with open(vfs_containers_conf_fpath, "w") as file:
+            toml.dump(vfs_containers_conf_data, file)
+
+        with open(vfs_containers_storage_conf_fpath, "w") as file:
+            toml.dump(vfs_containers_storage_conf_data, file)
+
+        oci_environ = os.environ.copy()
+        oci_environ.update(
+            {
+                "CONTAINERS_CONF": str(vfs_containers_conf_fpath),
+                "CONTAINERS_STORAGE_CONF": str(vfs_containers_storage_conf_fpath),
+            }
+        )
+
+        yield {
+            "container_engine": "podman",
+            "docker_image": DEFAULT_IMAGE,
+            "env": oci_environ,
+        }
```

Comment:
I don't understand all the complexities here! But do we need to run all the tests with this configuration? Or would just one be fine?

Reply:
I recall encountering instances where the logic broke one but not the other. Is there some mechanism to mark a test as "extra" or "slow?" It might be reasonable to move that into a group that isn't tested regularly on the CI, if runtime is the concern. If I find an instance where it is necessary I'll describe it in comments, otherwise I'm not opposed to removing it as long as my use case isn't breaking (which is what it's testing). I'll send in an issue if it does :)

Reply:
I moved the logic for setting up the environs and paths for the VFS test into it's own function. This should make it easier to read what configurations `basis_container_kwargs` generates.